core-api/src/main/java/com/optimizely/ab/bucketing/Bucketer.java

```diff
@@ -102,6 +109,19 @@ private Variation bucketToVariation(@Nonnull Experiment experiment,
                                         @Nonnull String userId) {
         // "salt" the bucket id using the experiment id
         String combinedBucketId = userId + experiment.getId();
+
+        // If a persistent bucketer instance is present then check it for a persisted variation
+        if (userExperimentRecord != null) {
+            String variationKey = userExperimentRecord.lookup(userId, experiment.getKey());
+            if (variationKey != null) {
+                logger.info("Returning previously activated variation \"{}\" from user experiment registry.", variationKey);
+                // A variation is stored for this combined bucket id
+                return projectConfig.getExperimentIdMapping().get(experiment.getId()).getVariationKeyToVariationMap().get(variationKey);
```

Comment:
line too long, put separate parts of function chain on different lines
Also since experiment id is used on line 111 as well for the `combinedBucketId` you can store it in a variable.